es6-shim.js

```diff
@@ -69,6 +69,7 @@
   var _some = Function.call.bind(Array.prototype.some);
 
   var defineProperty = function (object, name, value, force) {
+    object = object || {};
     if (!force && name in object) { return; }
```

Comment:
reassigning arguments deoptimizes the function. instead:
```suggestion
    object = object || {};
    if (!object || (!force && name in object)) { return; }
```

altho i'd rather fix all the callers of `defineProperty` and ensure they're not passing a non-object here.